### PR TITLE
Update documentation and move pre-requisites to the main README.md

### PR DESCRIPTION
**Pull request by pmalarme**
This pull request includes updates to the `README.md` files to improve clarity and provide additional resources for setting up the environment. The most important changes include adding a prerequisites section, enhancing the documentation with links to relevant resources, and removing redundant information.

Documentation updates:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R16-R24): Added a prerequisites section to guide users on setting up the necessary tools and environment.
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L33-R49): Enhanced the resource list with links to relevant Azure documentation for easier access.

Redundant information removal:

* [`src/ai-image-processing-service/README.md`](diffhunk://#diff-29bfebf051e2581119c5f43637bfa28e2c182678c7e27312569a4ec494c4e762L13-L21): Removed the duplicated prerequisites section to avoid redundancy and maintain consistency.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,12 +39,14 @@ To run the AI Shop locally, you need first to build the common module and then s
 The following resources will be created as represented in the diagram below:
 - Resource Group
 - Azure Container Registry
-- Azure Container Apps Environment
+- [Azure Container Apps Environment](https://learn.microsoft.com/en-us/azure/container-apps/)
```
Comment (SandraAhlgrimm) [inline README.md:42]:
```suggestion
- [Azure Container Apps Environment](https://learn.microsoft.com/azure/container-apps/)
```

File: README.md [hunk 2]
```diff
@@ -30,12 +39,14 @@ To run the AI Shop locally, you need first to build the common module and then s
...
+    - [Eureka Server for Spring Component](https://learn.microsoft.com/en-us/azure/container-apps/java-eureka-server-usage)
```
Comment (SandraAhlgrimm) [inline README.md:43]:
```suggestion
    - [Eureka Server for Spring Component](https://learn.microsoft.com/azure/container-apps/java-eureka-server-usage)
```

File: README.md [hunk 3]
```diff
@@ -30,12 +39,14 @@ To run the AI Shop locally, you need first to build the common module and then s
...
+    - [Admin for Spring Component](https://learn.microsoft.com/en-us/azure/container-apps/java-admin-for-spring-usage)
```
Comment (SandraAhlgrimm) [inline README.md:44]:
```suggestion
    - [Admin for Spring Component](https://learn.microsoft.com/azure/container-apps/java-admin-for-spring-usage)
```

File: README.md [hunk 4]
```diff
@@ -30,12 +39,14 @@ To run the AI Shop locally, you need first to build the common module and then s
...
 - Azure Blob Storage Account
 - Azure Key Vault for the `item-category-service` Azure OpenAI API Key secret
 - User-assigned Managed Identity to pull the images from the Azure Container Registry
 - Azure OpenAI
-- Azure Container Apps
+- [Azure Container Apps with Java Runtime](https://learn.microsoft.com/en-us/azure/container-apps/java-metrics)
```
Comment (SandraAhlgrimm) [inline README.md:49]:
```suggestion
- [Azure Container Apps with Java Runtime](https://learn.microsoft.com/azure/container-apps/java-metrics)
```

File: README.md [hunk 5]
```diff
@@ -13,6 +13,15 @@ The functional architecture of the application is as follows:
 
 ![Functional Architecture](./media/functional-architecture.png)
 
+## Prerequisites
+
+Easiest way to start is to [deploy a GitHub Codespace](https://github.com/codespaces/new/Azure-Samples/java-ai) as it contains all the prerequisites.
+
+If you deploy locally, you need to have the following installed:
+- [Java 17+](https://learn.microsoft.com/en-us/java/openjdk/download)
```
Comment (SandraAhlgrimm) [inline README.md:21]:
```suggestion
- [Java 17+](https://learn.microsoft.com/java/openjdk/download)
```

File: README.md [hunk 6]
```diff
@@ -13,6 +13,15 @@ The functional architecture of the application is as follows:
...
+- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
```
Comment (SandraAhlgrimm) [inline README.md:22]:
```suggestion
- [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli)
```

File: README.md [hunk 7]
```diff
@@ -13,6 +13,15 @@ The functional architecture of the application is as follows:
 
 ![Functional Architecture](./media/functional-architecture.png)
 
+## Prerequisites
+
+Easiest way to start is to [deploy a GitHub Codespace](https://github.com/codespaces/new/Azure-Samples/java-ai) as it contains all the prerequisites.
```
Comment (SandraAhlgrimm) [inline README.md:18]: I think they'll have to Fork it first.

```suggestion
Easiest way to start is to Fork and [open the repository in a GitHub Codespace](https://github.com/codespaces/new/Azure-Samples/java-ai) as it contains all the prerequisites.
```

File: README.md [hunk 8]
```diff
@@ -13,6 +13,15 @@ The functional architecture of the application is as follows:
...
+
+If you deploy locally, you need to have the following installed:
```
Comment (SandraAhlgrimm) [inline README.md:20]:
```suggestion
If you develop locally, you need to have the following installed:
```

**Review by SandraAhlgrimm — CHANGES_REQUESTED** [review]
Thank you.

Let's remove the localization from the links, so people get the documentation with their settings or based on their country.
And I've suggested a few minor changes.

**Comment by pmalarme** [comment]
> Thank you.
> 
> Let's remove the localization from the links, so people get the documentation with their settings or based on their country. And I've suggested a few minor changes.

Done

**Comment by SandraAhlgrimm** [comment]
Thank you!

**Review by SandraAhlgrimm — APPROVED** [review]

**Pull request merged**